File: Selenium/Activities/Activity15.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity15 {

	public static void main(String[] args) {

			WebDriver driver = new FirefoxDriver();
			WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(10));
			
			driver.get("https://training-support.net/webelements/dynamic-attributes");
			System.out.println("The page title is : "+ driver.getTitle());
			
			WebElement txtFullName = driver.findElement(By.xpath("//input[@placeholder='Full name']"));
			WebElement txtEmail = driver.findElement(By.xpath("//input[@placeholder='Email Address']"));
			WebElement txtEventTime = driver.findElement(By.xpath("//input[@data-testid='event-date']"));
			WebElement txtAdditionalDetails = driver.findElement(By.xpath("//textarea[contains(@name,'additional-details')]"));
			WebElement btnSubmit = driver.findElement(By.xpath("//button[text()='Submit']"));
			
			WebElement btnConfirmMsg =  driver.findElement(By.xpath("//h3[@id='action-confirmation']"));
			
			txtFullName.sendKeys("Rakhi Das");
			txtEmail.sendKeys("rakhids08@gmail.com");
			txtEventTime.sendKeys("2026-08-08");
			txtAdditionalDetails.sendKeys("Home Party");
			
			btnSubmit.click();
			
			String msg = wait.until(ExpectedConditions.visibilityOf(btnConfirmMsg)).getText();
			System.out.println("Status message is : "+msg);
			driver.quit();

	}

}

File: Selenium/Activities/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-attributes")
    print(f"The page title is : {driver.title}")

    wait = WebDriverWait(driver,10)

    txtFullName = driver.find_element(By.XPATH,"//input[@placeholder='Full name']")
    txtEmail = driver.find_element(By.XPATH,"//input[@placeholder='Email Address']")
    txtEventTime = driver.find_element(By.XPATH,"//input[@data-testid='event-date']")
    txtAdditionalDetails = driver.find_element(By.XPATH,"//textarea[contains(@name,'additional-details')]")
    btnSubmit = driver.find_element(By.XPATH,"//button[text()='Submit']")
    btnConfirmMsg =  driver.find_element(By.XPATH,"//h3[@id='action-confirmation']")
			
    txtFullName.send_keys("Rakhi Das")
    txtEmail.send_keys("rakhids08@gmail.com")
    txtEventTime.send_keys("2026-08-08")
    txtAdditionalDetails.send_keys("Home Party")
			
    btnSubmit.click();
			
    msg = wait.until(EC.visibility_of(btnConfirmMsg)).text
    print(f"Status message is : {msg}")
    driver.quit()